app/services/transaction_service/paypal_events.rb

```diff
@@ -118,8 +118,11 @@ def pending_ext?(pending_reason)
   end
 
   def pending_ext_to_paid(tx)
-    MarketplaceService::Transaction::Command.transition_to(tx[:id], :paid)
+    # Commission charge is synchronous and must complete before we
+    # transition to paid so that we have the full payment info
+    # available at the time we send payment receipts.
     TransactionService::Transaction.charge_commission(tx[:id])
+    MarketplaceService::Transaction::Command.transition_to(tx[:id], :paid)
```

Comment:
Why did you want to change the order here? If we follow the current logic charging a commission should be a reaction to transaction having moved to paid, so it too should move out of the paypal event handler similar to harmony notifications.